src/test/eventSchemas/providers/schemasDataProvider.test.ts

```diff
@@ -0,0 +1,98 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+import { SchemasDataProvider } from '../../../eventSchemas/providers/schemasDataProvider'
+import { MockSchemaClient } from '../../shared/clients/mockClients'
+import { asyncGenerator } from '../../utilities/collectionUtils'
+
+describe('getRegistries', () => {
+    let sandbox: sinon.SinonSandbox
+    beforeEach(() => {
+        sandbox = sinon.createSandbox()
+    })
+
+    afterEach(() => {
+        sandbox.restore()
+    })
+
+    const TEST_REGION = 'testRegion'
+    const TEST_REGISTRY = 'testRegistry'
+    const TEST_REGISTRY2 = 'testRegistry2'
+
+    const TEST_SCHEMA = 'testSchema'
+    const TEST_SCHEMA2 = 'testSchema2'
+
+    const schemaClient = new MockSchemaClient(TEST_REGION)
+
+    it('should return registries for given region', async () => {
+        const registrySummary1 = { RegistryArn: 'arn:aws:registry/' + TEST_REGISTRY, RegistryName: TEST_REGISTRY }
+        const registrySummary2 = { RegistryArn: 'arn:aws:registry/' + TEST_REGISTRY2, RegistryName: TEST_REGISTRY2 }
+
+        sandbox.stub(schemaClient, 'listRegistries').returns(asyncGenerator([registrySummary1, registrySummary2]))
+
+        const registryNames = await SchemasDataProvider.getInstance().getRegistries(TEST_REGION, schemaClient)
+
+        assert.ok(registryNames.length === 2, 'Should be two registries')
+        assert.strictEqual(registryNames[0], TEST_REGISTRY, 'TEST_REGISTRY name should match')
+        assert.strictEqual(registryNames[1], TEST_REGISTRY2, 'TEST_REGISTRY2 name should match')
+    })
+
+    it('should retain results once it is queried ', async () => {
+        const cachedResults = SchemasDataProvider.getInstance().getCachedRegionMap()
```

Comment:
These tests appear to be side-effect based. The tests should test and verify in isolation of one another.

You should be able to instantiate SchemasDataProvider objects for each test. If each test has the same setup, that can be placed into a beforeEach block, which still gives each test its own freshly instantiated object.

Applies to whole file